src/main/java/de/tum/in/www1/artemis/web/rest/LectureResource.java

```diff
@@ -171,35 +174,47 @@ public LectureResource(LectureRepository lectureRepository, LectureService lectu
         if (!authCheckService.isAtLeastStudentInCourse(course, user)) {
             return forbidden();
         }
+        lecture = filterLectureContentForUser(lecture, user);
+
+        return ResponseEntity.ok(lecture);
+    }
+
+    private Lecture filterLectureContentForUser(Lecture lecture, User user) {
         lecture = lectureService.filterActiveAttachments(lecture, user);
 
-        // to make sure that the information provided for lecture units is equal to the one in the course dashboard
-        // ToDo Improve performance by constructing a more precise call
-        course = this.courseResource.getCourseForDashboard(course.getId());
-        Set<Exercise> exercisesUserIsAllowedToSee = course.getExercises();
+        // The Objects::nonNull is needed here because the relationship lecture -> lecture units is ordered and
+        // hibernate sometimes adds nulls to in the list of lecture units to keep the order
```

Comment:
`to in` -> `into`

Reply:
Changed in 1731fed7a02c651ee44973f34f3d276b1aac5cd1